clippy_lints/src/utils/internal_lints.rs

```diff
@@ -304,3 +324,22 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for OuterExpnDataPass {
         }
     }
 }
+
+declare_lint_pass!(ProduceIce => [PRODUCE_ICE]);
+
+impl EarlyLintPass for ProduceIce {
+    fn check_fn(&mut self, _: &EarlyContext<'_>, fn_kind: FnKind<'_>, _: &ast::FnDecl, _: Span, _: ast::NodeId) {
+        if is_trigger_fn(fn_kind) {
+            panic!("Testing the ICE message");
+        }
+    }
+}
+
+fn is_trigger_fn(fn_kind: FnKind<'_>) -> bool {
+    match fn_kind {
+        FnKind::ItemFn(ident, ..) | FnKind::Method(ident, ..) => {
+            ident.name.as_str() == "should_trigger_an_ice_in_clippy"
```

Comment:
Though Clippy internal lints are never emitted by Clippy, they still run all the time. I don't think that anyone would name a function `should_trigger_an_ice_in_clippy`, but I suggest to change this name to one of the following:

1. `should_trigger_an_ice_when_clippy_is_run`
2. Make it a Clippy easteregg and give it a cool name :sunglasses: `it_looks_like_you_are_trying_to_kill_clippy` or similar.

Reply:
ooh i like the easter egg idea